pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -530,11 +537,23 @@ func (h *Health) checkWorkers(
 	return &c, nil
 }
 
+var constraintGardenerGreaterEqual190 *semver.Constraints
+
+func init() {
+	var err error
+	constraintGardenerGreaterEqual190, err = semver.NewConstraint(">= 1.90-0")
+	utilruntime.Must(err)
+}
+
 // ComputeRequiredMonitoringStatefulSets returns names of monitoring statefulsets based on the given shoot.
-func ComputeRequiredMonitoringStatefulSets(wantsAlertmanager bool) sets.Set[string] {
+func ComputeRequiredMonitoringStatefulSets(wantsAlertmanager bool, gardenerVersion *semver.Version) sets.Set[string] {
 	var requiredMonitoringStatefulSets = sets.New(v1beta1constants.StatefulSetNamePrometheus)
 	if wantsAlertmanager {
-		requiredMonitoringStatefulSets.Insert(v1beta1constants.StatefulSetNameAlertManager)
+		if constraintGardenerGreaterEqual190.Check(gardenerVersion) {
+			requiredMonitoringStatefulSets.Insert("alertmanager-shoot")
+		} else {
+			requiredMonitoringStatefulSets.Insert(v1beta1constants.StatefulSetNameAlertManager)
+		}
```

Comment:
May be you can add TODO to drop it after a release or two.

Reply:
https://github.com/gardener/gardener/pull/9313 will drop this code already with the next release and check the `ManagedResource` instead